ta/remoteproc/remoteproc_core.c

```diff
@@ -0,0 +1,1079 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2023, STMicroelectronics
+ */
+
+#include <elf_parser.h>
+#include <remoteproc_pta.h>
+#include <string.h>
+#include <sys/queue.h>
+#include <ta_remoteproc.h>
+#include <tee_internal_api.h>
+#include <tee_internal_api_extensions.h>
+#include <types_ext.h>
+#include <utee_defines.h>
+
+/*
+ * The remoteproc Trusted Application is in charge of authenticating, loading
+ * images signed by the scripts/sign_rproc_fw.py. The TA is also in charge of
+ * starting and stopping the remote processor.
+ * The structure of the signed image is:
+ *
+ *                   -----+-------------+
+ *                  /     |    Magic    |  32-bit word, magic value equal to
+ *                 /      +-------------+  0x3543A468
+ *                /       +-------------+
+ *               /        |   version   |  32-bit word, version of the format
+ *              /         +-------------+
+ * +-----------+          +-------------+
+ * |   Header  |          |  TLV size   |  32-bit word, size of the TLV
+ * +-----------+          +-------------+  (aligned on 64-bit), in bytes.
+ *              \         +-------------+
+ *               \        |  sign size  |  32-bit word, size of the signature
+ *                \       +-------------+  (aligned on 64-bit), in bytes.
+ *                 \      +-------------+
+ *                  \     | images size |  32-bit word, size of the images to
+ *                   -----+-------------+  load (aligned on 64-bit), in bytes.
+ *
+ *                        +-------------+  Information used to authenticate the
+ *                        |     TLV     |  images and boot the remote processor,
+ *                        |             |  stored in Type-Length-Value format.
+ *                        +-------------+  'Type' and 'Length' are 32-bit words.
+ *
+ *                        +-------------+
+ *                        | Signature   |   Signature of the header and the TLV.
+ *                        +-------------+
+ *
+ *                        +-------------+
+ *                        |   Firmware  |
+ *                        |    image 1  |
+ *                        +-------------+
+ *                               ...
+ *                        +-------------+
+ *                        |   Firmware  |
+ *                        |    image n  |
+ *                        +-------------+
+ */
+
+/* Firmware state */
+enum remoteproc_state {
+	REMOTEPROC_OFF,
```

Comment:
explicit `= 0` since it is the expected default state when context is allocated (and zero initiallized)?